### ISSUE TITLE
Motion blur filter outputs a broken image

### ISSUE THREAD

**Issue by Umio-Yasuno** [GPUOpen-LibrariesAndSDKs/RadeonImageFilter#8]
Hello.  
I made [rif-cmd](https://github.com/Umio-Yasuno/rif-cmd) for the purpose of testing RIF.  

Motion Blur filter will output a broken image depending on the image size and extension.
If the source is a 640x360 JPEG image, the filter will always output a broken image.

Is this a bug in RIF?

(The images were taken by me. I declare it to be in the public domain.)  

Source (PNG, 640x360) :
![source png](https://user-images.githubusercontent.com/53935716/104842864-ee732280-590a-11eb-8437-f0a17d9b19ec.png)

Radius [5] (PNG, 640x360) :
![test_radius_5](https://user-images.githubusercontent.com/53935716/104842902-2712fc00-590b-11eb-97a1-affe2441c716.png)
[rif-radius_5.log](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5826066/rif-radius_5.log)

Radius [10] (PNG, 640x360) :
![test_radius_10](https://user-images.githubusercontent.com/53935716/104842921-5164b980-590b-11eb-96f1-b702bb8ebd49.png)
[rif-radius_10.log](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5826065/rif-radius_10.log)

OS: Debian GNU/Linux bullseye/sid
CPU: Ryzen 5 2600
GPU: Radeon RX 560 4GB (Polaris11, gfx803)
OpenCL: OpenCL 2.0 AMD-APP (3212.0)

**Comment by BenjaminCoquelle** [comment]
can you create a RIF trace please?

**Comment by Umio-Yasuno** [comment]
Radius [5]:
[rif-trace-1610910843-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5826663/rif-trace-1610910843-1.6.2.0xbdf2aac0.tar.gz)

Radius [10]: 
[rif-trace-1610911169-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5826662/rif-trace-1610911169-1.6.2.0xbdf2aac0.tar.gz)

**Comment by BenjaminCoquelle** [comment]
thanks we will look at it asap

**Comment by stmuxa** [comment]
Hi @Umio-Yasuno 
I confirm a bug in motion filter here.
You can either use a temporal workaround
```
rifImageFilterSetParameter1u(filter_MOTION_BLUR, "radius", 4u));
rifImageFilterSetParameter1u(filter_MOTION_BLUR, "radius", 5u));
```
or wait for next version.

**Comment by Umio-Yasuno** [comment]
Thanks, @stmuxa
In the case of PNG, it will output an image with no problem.  

However, in the case of JPEG images, it keeps outputting broken images.  
This may be a different problem, as other filters do that as well.

Source (JPEG, 640x360) :
![](https://user-images.githubusercontent.com/53935716/104904231-4a4db200-59c4-11eb-98d1-6610e80fd26f.jpg)

Bloom (deafult value)
[rif-trace-1610965982-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5829608/rif-trace-1610965982-1.6.2.0xbdf2aac0.tar.gz)

Motion Blur (deafult value)
[rif-trace-1610966462-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5829638/rif-trace-1610966462-1.6.2.0xbdf2aac0.tar.gz)

**Comment by stmuxa** [comment]
RadeonImageFilter has nothing to do with image formats like PNG and JPEG, it works with raw images only.
I extracted in/out images from the traces you sent, and they does not seen correpted.

> Bloom (deafult value)
> [rif-trace-1610965982-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5829608/rif-trace-1610965982-1.6.2.0xbdf2aac0.tar.gz)

input
![image_0](https://user-images.githubusercontent.com/10169674/104908662-d7c6dd00-5997-11eb-8254-691b7a816cd7.jpg)
output
![image_1](https://user-images.githubusercontent.com/10169674/104908668-d8f80a00-5997-11eb-9342-885c05a104e2.jpg)


> Motion Blur (deafult value)
> [rif-trace-1610966462-1.6.2.0xbdf2aac0.tar.gz](https://github.com/GPUOpen-LibrariesAndSDKs/RadeonImageFilter/files/5829638/rif-trace-1610966462-1.6.2.0xbdf2aac0.tar.gz)

input
![image_0](https://user-images.githubusercontent.com/10169674/104908505-a3532100-5997-11eb-814a-73d691318180.jpg)

output
![image_1](https://user-images.githubusercontent.com/10169674/104908510-a3ebb780-5997-11eb-9beb-73debd8fce1b.jpg)

**Comment by Umio-Yasuno** [comment]
Sorry, there may be a problem with the implementation of the image output process.  
I will try another way.

**Comment by Umio-Yasuno** [comment]
It's solved!  
By setting the `type` of `rif_image_desc` to `RIF_COMPONENT_TYPE_UINT8` , `rif-cmd` was able to output JPEG images without any problems.

**Comment by Umio-Yasuno** [comment]
I have confirmed the fix in v1.7.0 .
Thank you.
